py-polars/polars/dataframe/frame.py

```diff
@@ -1227,11 +1235,11 @@ def __dataframe__(
                 "pyarrow>=11.0.0 is required for converting a Polars dataframe to a"
                 " dataframe interchange object."
             )
-        if not allow_copy and Categorical in self.schema.values():
-            raise NotImplementedError(
-                "Polars does not offer zero-copy conversion to Arrow for categorical"
-                " columns. Set `allow_copy=True` or cast categorical columns to"
-                " string first."
+        if not allow_copy and Categorical in unpack_dtypes(*self.dtypes):
```

Comment:
Ok, I double checked the code. We are zero copy if:

The categorical is built locally.

We are copy if:

The categorical is built under a global string cache. 

So let's go forward with this for now. I think we must modify the message, indicating that we cannot guarantee currently.. We can add method to get the local/global information and be less conservative.

Reply:
I updated the message to say _"Polars can not currently guarantee zero-copy conversion to Arrow for categorical columns"_. That should be more correct.

I don't think we have a good way to check whether for a column whether it has been built locally or globally, right?

Or maybe we can just call `to_arrow` and raise an error when it tries to do any copying (we can give it a similar argument like `allow_copy`)?

For now, I think this is fine to merge. We can improve to allow zero-copy for local categoricals. And hopefully we'll have our own implementation soon and this won't matter anymore.

Reply:
> I don't think we have a good way to check whether for a column whether it has been built locally or globally, right?

It is information in the datatype, but it isn't exposed yet. I can create a function for that, but that should not be a blocker for this PR.